### ISSUE TITLE
Gradle errors with -Xms above 512m

### ISSUE THREAD

**Issue by anderssv** [gradle/gradle#8182]
On upgrading to Gradle 5.1 (from 4.10) it fails whenever I use a -Xms setting that is higher than 512m . 
This is not something that we strictly have to set, but it is the default on performance-l dynos on Heroku. And thus our CI.

`$ export JAVA_TOOL_OPTIONS="-XX:+UseContainerSupport -Xms513m -Xmx12g -Dfile.encoding=UTF-8" && ./gradlew clean build

Picked up JAVA_TOOL_OPTIONS: -XX:+UseContainerSupport -Xms513m -Xmx12g -Dfile.encoding=UTF-8
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Unable to start the daemon process.
This problem might be caused by incorrect configuration of the daemon.
For example, an unrecognized jvm option is used.
Please refer to the user guide chapter on the daemon at https://docs.gradle.org/5.1/userguide/gradle_daemon.html
Please read the following process output to find out more:
-----------------------
Picked up JAVA_TOOL_OPTIONS: -XX:+UseContainerSupport -Xms513m -Xmx12g -Dfile.encoding=UTF-8
Error occurred during initialization of VM
Initial heap size set to a larger value than the maximum heap size

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
`

### Expected Behavior
Normal build and execution of tests

### Current Behavior
Fails with the above error

### Context
Running and building our application with default settings on Heroku. We don't normally use -Xms in other contexts.

### Steps to Reproduce (for bugs)
Should fail on any project with Java tests with the command:

`export JAVA_TOOL_OPTIONS="-XX:+UseContainerSupport -Xms513m -Xmx12g -Dfile.encoding=UTF-8" && ./gradlew clean build`

### Your Environment

Failing locally and on Heroku with Java 11.

**Comment by ljacomet** [comment]
Indeed, so that's a priority issue. You will have to repeat the `-Xmx` at a higher level, like in `JAVA_OPTS` or `GRADLE_OPTS` so that the value is not overwritten by the Gradle launch script.

**Comment by anderssv** [comment]
Not sure I got you guys. I see it's closed, so that means we should be using the JAVA_OPTS instead for setting this? JAVA_TOOL_OPTIONS is meant for options to the Gradle daemon? Or is this something that will change? Thanks. :)

**Comment by GFriedrich** [comment]
```JAVA_TOOL_OPTIONS``` is a generic approach to set common arguments to any JVM (e.g. Gradle daemon but of course not limited to that).
As Gradle sets some default parameters for its daemon ("-Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError") that will overrule the ```JAVA_TOOL_OPTIONS``` parameter, the JVM won't start.
You can override the Gradle defaults by either using the ```JAVA_OPTS``` or ```GRADLE_OPTS``` environment variable.

**Comment by anderssv** [comment]
Thanks for the clarification. :) 

Not sure what the custom is as JAVA_TOOL_OPTIONS is quite new to me. At least I think it breaks the Heroku buildpack way of doing things. (https://github.com/heroku/heroku-buildpack-gradle) I'll create a report there. :) 

Thanks again.

**Comment by jkutner** [comment]
@ljacomet If gradle is setting itself to a fixed max heap (64M) then it should also set the min heap so that it works with standard JVM mechanisms like `_JAVA_OPTIONS` and `JAVA_TOOL_OPTIONS`. The `JAVA_OPTS` env var is only a convention that's used by lots of frameworks.

Thus, I believe this is a bug in Gradle. @ljacomet would you accept a PR if try to the problem in this repo?

**Comment by ljacomet** [comment]
I understand the feedback and indeed if we were to specify `-Xms` as well, the execution would not have failed. However it would still have ignored the values set by the user from `JAVA_TOOL_OPTIONS` as they are prepended and thus overwritten by the `gradlew` script's own values.
I wonder how this issue gets resolved for other tools that have an opinion on JVM memory sizing?
